### PR TITLE
Left-to-right test

### PR DESCRIPTION
**Pull request by pjuftring**
This tests, if all arithmetic instructions evaluate their operands from left to right.

At the moment, I try to implement as many tests mentioned in TestingTodo.md as I can. I hope, that's ok :)
Should I make a pull request for each test, or wait until I have a bunch of tests and make a Pull Request for all of them together?

### CODE REVIEW

File: ml-proto/test/left-to-right.wast [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  (func $i32_right (result i32) (i32.store8 (i32.const 0) (i32.const 2)) (i32.const 141))
+  (func $i64_left (result i64) (i32.store8 (i32.const 0) (i32.const 1)) (i64.const 163))
+  (func $i64_right (result i64) (i32.store8 (i32.const 0) (i32.const 2)) (i64.const 155))
+  (func $f32_left (result f32) (i32.store8 (i32.const 0) (i32.const 1)) (f32.const 124))
+  (func $f32_right (result f32) (i32.store8 (i32.const 0) (i32.const 2)) (f32.const 145))
+  (func $f64_left (result f64) (i32.store8 (i32.const 0) (i32.const 1)) (f64.const 163))
+  (func $f64_right (result f64) (i32.store8 (i32.const 0) (i32.const 2)) (f64.const 164))
+
+
+  (func $i32_add (result i32) (call $reset)(i32.add (call $i32_left)(call $i32_right))(call $get))
```
Comment (sunfishcode) [inline ml-proto/test/left-to-right.wast]: For consistency with the overall style of the tests in this directory, please put spaces between operands.

**Comment by sunfishcode** [comment]
Other than one comment above, lgtm. Implementing tests in TestingTodo.md is great! Submitting test cases either individually or in groups is fine with me.

FYI, I'm also writing some tests -- https://github.com/WebAssembly/spec/pull/113 -- so if there are specific things you're interested in working on, let me know and we can coordinate and avoid duplicating effort.

**Comment by sunfishcode** [comment]
lgtm; thanks!

**Comment by pjuftring** [comment]
@sunfishcode You said, I should let you know on which tests I'm working at the moment. So here is a list of tests I'm interested in/working on:
(I'm not sure if this is the right place to post this, should I have opened an "Issue" for that?)

- test that call and store operands are evaluated left-to-right too
- test that page_size returns a power of 2
- test that one can clobber the entire contents of the linear memory without corrupting: call stack, global variables, local variables, program execution.
- test that converting int32 to float and back isn't folded away
- test that converting int64 to double and back isn't folded away

**Comment by sunfishcode** [comment]
Great! I've now created #132 to serve as a central place for discussing writing tests for the remainder of TestingTodo.md going forward.

**Pull request merged**